avs_core/convert/intel/convert_audio_sse.cpp

```diff
@@ -0,0 +1,565 @@
+    );
+    outv[2] = _mm_or_si128(
+      _mm_shuffle_epi8(inv[2], mask[4]),
+      _mm_shuffle_epi8(inv[3], mask[5])
+    );
+
+    _mm_storeu_si128(reinterpret_cast<__m128i *>(out8), outv[0]); out8 += 16;
+    _mm_storeu_si128(reinterpret_cast<__m128i *>(out8), outv[1]); out8 += 16;
+    _mm_storeu_si128(reinterpret_cast<__m128i *>(out8), outv[2]); out8 += 16;
+  }
+  // clang-format on
+}
+
+SSSE3 void convert24To32_SSSE3(void *inbuf, void *outbuf, int count) {
+  auto in8 = reinterpret_cast<int8_t *>(inbuf);
+  auto out8 = reinterpret_cast<int8_t *>(outbuf);
+  auto out = reinterpret_cast<int32_t *>(outbuf);
+
+  const int c_loop = count & ~15;
+
+  for (int i = c_loop; i < count; i++) {
+    out8[i * 4] = 0;
+    out8[i * 4 + 1] = in8[i * 3 + 0];
+    out8[i * 4 + 2] = in8[i * 3 + 1];
+    out8[i * 4 + 3] = in8[i * 3 + 2];
+  }
+
+  __m128i inv[3], outv[4], mask[6];
+  // clang-format off
+  mask[0] = _mm_set_epi8(
+    11, 10,  9, -1,
+     8,  7,  6, -1,
+     5,  4,  3, -1,
+     2,  1,  0, -1);
+  mask[1] = _mm_set_epi8(
+    -1, -1, -1, -1,
+    -1, -1, -1, -1,
+    -1, -1, 15, -1,
+    14, 13, 12, -1);
+  mask[2] = _mm_set_epi8(
+     7,  6,  5, -1,
+     4,  3,  2, -1,
+     1,  0, -1, -1,
+    -1, -1, -1, -1);
+  mask[3] = _mm_set_epi8(
+    -1, -1, -1, -1,
+    -1, 15, 14, -1,
+    13, 12, 11, -1,
+    10,  9,  8, -1);
+  mask[4] = _mm_set_epi8(
+     3,  2,  1, -1,
+     0, -1, -1, -1,
+    -1, -1, -1, -1,
+    -1, -1, -1, -1);
+  mask[5] = _mm_set_epi8(
+    15, 14, 13, -1,
+    12, 11, 10, -1,
+     9,  8,  7, -1,
+     6,  5,  4, -1);
+
+  for (int i = 0; i < c_loop; i += 16) {
+    inv[0] = _mm_loadu_si128(reinterpret_cast<const __m128i *>(in8)); in8 += 16;
+    inv[1] = _mm_loadu_si128(reinterpret_cast<const __m128i *>(in8)); in8 += 16;
+    inv[2] = _mm_loadu_si128(reinterpret_cast<const __m128i *>(in8)); in8 += 16;
+
+    outv[0] = _mm_shuffle_epi8(inv[0], mask[0]);
+    outv[1] = _mm_or_si128(
+      _mm_shuffle_epi8(inv[0], mask[1]),
+      _mm_shuffle_epi8(inv[1], mask[2])
+    );
+    outv[2] = _mm_or_si128(
+      _mm_shuffle_epi8(inv[1], mask[3]),
+      _mm_shuffle_epi8(inv[2], mask[4])
+    );
+    outv[3] = _mm_shuffle_epi8(inv[2], mask[5]);
+
+    _mm_storeu_si128(reinterpret_cast<__m128i *>(out), outv[0]); out += 4;
+    _mm_storeu_si128(reinterpret_cast<__m128i *>(out), outv[1]); out += 4;
+    _mm_storeu_si128(reinterpret_cast<__m128i *>(out), outv[2]); out += 4;
+    _mm_storeu_si128(reinterpret_cast<__m128i *>(out), outv[3]); out += 4;
+  }
+  // clang-format on
+}
+
+SSSE3 void convert24To16_SSSE3(void *inbuf, void *outbuf, int count) {
+  auto in8 = reinterpret_cast<int8_t *>(inbuf);
+  auto out8 = reinterpret_cast<int8_t *>(outbuf);
+  auto out = reinterpret_cast<int16_t *>(outbuf);
+
+  const int c_loop = count & ~15;
+
+  for (int i = c_loop; i < count; i++) {
+    out8[i * 2 + 0] = in8[i * 3 + 1];
+    out8[i * 2 + 1] = in8[i * 3 + 2];
+  }
+
+  __m128i inv[3], outv[2], mask[4];
+  // clang-format off
```

Comment:
Is that hint to prevent autoformatter being too smart?

Reply:
Yes. I configured clang-format to give the code a clean style. To make sure it doesn't wipe the meaningful line wrapping I put that comment there.